crates/router/src/core/payments/operations.rs

```diff
@@ -41,32 +41,33 @@ use crate::{
     },
 };
 
-pub type BoxedOperation<'a, F, T> = Box<dyn Operation<F, T> + Send + Sync + 'a>;
+pub type BoxedOperation<'a, F, T, D> = Box<dyn Operation<F, T, Data = D> + Send + Sync + 'a>;
 
 pub trait Operation<F: Clone, T>: Send + std::fmt::Debug {
-    fn to_validate_request(&self) -> RouterResult<&(dyn ValidateRequest<F, T> + Send + Sync)> {
+    type Data;
+    fn to_validate_request(
+        &self,
+    ) -> RouterResult<&(dyn ValidateRequest<F, T, Self::Data> + Send + Sync)> {
```

Comment:
should validate request also have the PaymentData associated with it? The lifecycle of PaymentData starts from getTrackers right?

Reply:
We are returning `BoxedOperation` from `ValidateRequest` which needs the type information. So the type is added to `ValidateRequest` too. This is only from type stitching